app/src/main/java/swati4star/createpdf/activity/MainActivity.java

```diff
@@ -271,6 +272,10 @@ public void onBackPressed() {
                     .findFragmentById(R.id.content);
             if (currentFragment instanceof HomeFragment) {
                 checkDoubleBackPress();
+            } else if (mSharedPreferences.getBoolean(stateBottomSheet, true)) {
+                getSupportFragmentManager().beginTransaction().detach(currentFragment)
+                        .attach(currentFragment).commit();
+                mSharedPreferences.edit().putBoolean(stateBottomSheet, false).apply();
```

Comment:
I can't quite figure out why you're using shared preferences here. What if the application is closed with `true` saved inside the preference variable?

Reply:
i want to get the state of bottomsheet behaviour from all the fragment whether it is expanded or collapsed

Reply:
@srv-twry even if the application is closed with true saved inside the preference variable, it will just restart the fragment